core/primitives/authority.hpp

```diff
@@ -105,6 +105,12 @@ namespace kagome::primitives {
   Stream &operator>>(Stream &s, Authority &a) {
     return s >> a.id >> a.weight;
   }
+
+  /// Special type for vector of authorities
+  struct AuthorityList : public std::vector<Authority> {
+  		using std::vector<Authority>::vector;
+  };
+
```

Comment:
Destructor of base class will be called in any case. Virtual destructor needed for calling destructor of derived classes at delete over pointer/reference of base class.

Reply:
Right here AuthorityList have no additional member and logic. It doesn't matter.

Reply:
I know, I mean a potential leak, because of no guarantees that members will not be placed in this struct. You can do smth like this:
```c++
struct AuthorityList : public std::vector<Authority> {
  std::vector<Authority>::vector;
  std::string test;
  AuthorityList(std::string_view s) : test(s) { }
};
std::shared_ptr<std::vector<Authority>> ptr = std::make_shared<AuthorityList>("1231234234");
ptr.reset();
/// test.~string() won't be called